transport/src/main/java/io/netty/channel/PendingWriteQueue.java

```diff
@@ -112,7 +123,6 @@ public void removeAndFailAll(Throwable cause) {
         for (PendingWrite write = head; write != null; write = head) {
             head = tail = null;
             size = 0;
```

Comment:
Should `size=0` occur outside of the loop? And should we also be setting `bytes` here?